### PR TITLE
Add Python 3 compatibility

### PR DESCRIPTION
**Pull request by Pankrat**
These changes make abl.vpath usable with Python 2.7 and Python 3.x. 

The biggest/most critical changes were required in the memory file abstraction. It now always uses `BytesIO` internally and decodes & encodes on the `MemoryProxy` layer if the memory file is accessed in text mode. The reason for this design: The memory file instances stay the same even if the file is written as text but than read as a byte stream (this happens when you copy a memory file to regular file system file).

## Other changes

* On Linux, `lchmod` doesn't exist for chmodding symlinks. I've made the test Mac-only.
* Assertions have been changed to `assertEqual` to get more information from failing tests
* Wildcard imports have been replaced by regular imports (so that I can use static analysis)
* The backported zipfile library for old Python versions has been removed.
* This package is now tested on Travis CI.

Tests pass on Python 2.7, 3.5, 3.6 and 3.7. This has only been tested on Linux so far.

### CODE REVIEW

**Comment by ala-ableton** [comment]
@Pankrat this PR is super hard to review. Each commit contains a bunch of style changes that are not related to what the commit is supposedly doing. Could you please make another PR where you only run pyflakes and fix all the issues without doing any behavior change?

File: abl/vpath/base/fs.py [hunk 1]
```diff
@@ -153,12 +135,12 @@ def get_connection(self,
     def cleanup(self, force=False):
         import time
```
Comment (stedi67) [inline abl/vpath/base/fs.py]: `time` was already imported at the top.

File: abl/vpath/base/fs.py [hunk 2]
```diff
@@ -170,6 +152,7 @@ def cleaner(self):
         while True:
             now = time.time()
```
Comment (stedi67) [inline abl/vpath/base/fs.py]: I guess we can remove this line.

File: abl/vpath/base/fs.py [hunk 3]
```diff
@@ -984,8 +967,9 @@ def copy(self, source, dest, recursive=False, ignore=None,
                         if srcf.islink() and not followlinks:
                             self._copy_link(srcf, destf)
                         else:
-                            with nested(srcf.open('rb'), destf.open('wb') ) as (infs, outfs):
-                                shutil.copyfileobj(infs, outfs, 8192)
+                            with srcf.open('rb') as infs:
+                                with destf.open('wb') as outfs:
+                                    shutil.copyfileobj(infs, outfs, 8192)
```
Comment (stedi67) [inline abl/vpath/base/fs.py]: I'd prefer
```python
with srcf.open('rb') as infs, destf.open('wb') as outfs:
    shutil.copyfileobj(infs, outfs, 8192)
```

File: abl/vpath/base/memory.py [hunk 4]
```diff
@@ -77,6 +77,10 @@ def flush(self):
 
 
     def __str__(self):
```
Comment (stedi67) [inline abl/vpath/base/memory.py]: the `decode` transforms the bytestring into a unicode string. In python 2 we shouldn't do that, but just return the value. So, I guess, we need to redefine `__str__` for Python 2 as `__str__ = __bytes__`.

Comment (Pankrat) [reply abl/vpath/base/memory.py]: Good catch. I guess I'll add a version check here.

File: abl/vpath/base/localfs.py [hunk 5]
```diff
@@ -2,20 +2,26 @@
...
+
 from .fs import FileSystem, BaseUri, denormalize_path, URI
 from .exceptions import OperationIsNotSupportedOnPlatform
-from .os_abstraction import os
 
 from abl.util import Bunch, LockFile
 
+try:
+    lchmod = os.lchmod
+except AttributeError: # Python 3
```
Comment (Pankrat) [inline abl/vpath/base/localfs.py]: This is also raised in Python 2 on Linux.

**Comment by Pankrat** [comment]
@stedi67: My nosetests version silently skips the zip tests because the zip test module is executable - I'll chmod the test module and check if I can fix the tests.

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+### Added
+
+- Python 3.5, 3.6, 3.7 support
+
+### Changed
+
+- Memory files are always stored as binary streams and are encoded/decoded
+  based on the mode (r, rb, w, wb)
+- Switch to Python's standard zipfile libary
+- Python 2.5, 2.6 support
```
Comment (ala-ableton) [inline CHANGELOG.md]: @Pankrat did you mean to have a `### Removed` section or is there some word missing here?

Comment (Pankrat) [reply CHANGELOG.md]: Indeed, I'll fix the changelog on master. Thanks for spotting this.

**Pull request merged**